jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/MessageHandler.java

```diff
@@ -86,254 +85,210 @@ protected void onBinary(ByteBuffer message, Callback callback)
         };
     }
 
-    private static final Logger LOG = Log.getLogger(MessageHandler.class);
-
-    private final int factor;
+    protected static final Logger LOG = Log.getLogger(MessageHandler.class);
 
     private CoreSession coreSession;
-    private Utf8StringBuilder utf8StringBuilder = null;
-    private ByteBuffer binaryMessage = null;
+    private Utf8StringBuilder textMessageBuffer;
+    private ByteArrayOutputStream binaryMessageBuffer;
     private byte dataType = OpCode.UNDEFINED;
 
-    private int maxTextMessageSize = WebSocketConstants.DEFAULT_MAX_TEXT_MESSAGE_SIZE;
-    private int maxBinaryMessageSize = WebSocketConstants.DEFAULT_MAX_BINARY_MESSAGE_SIZE;
-
-    public MessageHandler()
-    {
-        this(3);
-    }
-
-    public MessageHandler(int factor)
-    {
-        this.factor = factor;
-    }
-
-    public int getMaxTextMessageSize()
-    {
-        return maxTextMessageSize;
-    }
-
-    public void setMaxTextMessageSize(int maxTextMessageSize)
-    {
-        this.maxTextMessageSize = maxTextMessageSize;
-    }
-
-    public int getMaxBinaryMessageSize()
+    public CoreSession getCoreSession()
     {
-        return maxBinaryMessageSize;
+        return coreSession;
     }
 
-    public void setMaxBinaryMessageSize(int maxBinaryMessageSize)
+    private Utf8StringBuilder getTextMessageBuffer()
```

Comment:
Good.   But if you wanted to, you could pass in an `int capacityHint` which could be determined as something like:
```java
getTextMessageBuffer(Math.min(maxMessageSize,frame.getSize()*frame.isFin()?1:2));
```

Reply:
Hmmm although that would then calculate the hint every frame..... so nah don't bother.... or just pass in framesize and do the calc only if needed.